### DISCUSSION TITLE
On five TS0503B, two are supported and 3 are not.

### DISCUSSION THREAD

**Discussion by bentouspam** [Koenkk/zigbee2mqtt#13375] [Devices]
Hello,

I Bought 5 Zigbee USB Ribon led controler TS0503B
![image](https://user-images.githubusercontent.com/101732505/182380761-58b13a08-d591-4ee2-b83a-eb3f626c980a.png)

on 5, 2 are OK:
![image](https://user-images.githubusercontent.com/101732505/182380941-2e44fa8e-7f5d-4422-89fa-4f57fbb3c0c3.png)
![image](https://user-images.githubusercontent.com/101732505/182381076-2bea5f9e-ebed-489b-95e0-3389cf320a29.png)

and 3 are not supported:
![image](https://user-images.githubusercontent.com/101732505/182381257-b1d16bca-229b-4dad-8a2a-621e91ab4964.png)
![image](https://user-images.githubusercontent.com/101732505/182381347-f37b66ca-e053-4988-a793-fd4f54eb35d0.png)
![image](https://user-images.githubusercontent.com/101732505/182381460-e3e31830-c554-49da-9b81-e4ef779dea69.png)

_TZ3210_trm3l2aw	RGB Ecrans
_TZ3000_i8l0nqdu	RGB Retro TV

_TZ3210_95txyzbx	Not supported
_TZ3210_95txyzbx	Not supported
_TZ3210_95txyzbx	Not supported

Is it a new variant ??? It's trendy right now...

[z2m log.txt](https://github.com/Koenkk/zigbee2mqtt/files/9243884/z2m.log.txt)
Z2m is version 1.27.0-1
Here are the logs during the first pairing (was working) and the second (that wasn't).

Is there a thing I can do ?

Regards,
Benoit.

**Comment by Koenkk** [comment]
Added!

Changes will be available in the dev branch in a few hours from now. (https://www.zigbee2mqtt.io/advanced/more/switch-to-dev-branch.html)

Until the next release you can use the following external converter.


```js
const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');
const ota = require('zigbee-herdsman-converters/lib/ota');
const tuya = require('zigbee-herdsman-converters/lib/tuya');
const e = exposes.presets;
const ea = exposes.access;

const definition = {
    fingerprint: [{modelID: 'TS0503B', manufacturerName: '_TZ3000_i8l0nqdu'},
        {modelID: 'TS0503B', manufacturerName: '_TZ3210_a5fxguxr'},
        {modelID: 'TS0503B', manufacturerName: '_TZ3210_778drfdt'},
        {modelID: 'TS0503B', manufacturerName: '_TZ3000_g5xawfcq'},
        {modelID: 'TS0503B', manufacturerName: '_TZ3210_trm3l2aw'},
        {modelID: 'TS0503B', manufacturerName: '_TZ3210_95txyzbx'},
        {modelID: 'TS0503B', manufacturerName: '_TZ3210_odlghna1'}],
    model: 'TS0503B',
    vendor: 'TuYa',
    description: 'Zigbee RGB light',
    extend: extend.light_onoff_brightness_color(),
    // Requires red fix: https://github.com/Koenkk/zigbee2mqtt/issues/5962#issue-796462106
    meta: {applyRedFix: true, enhancedHue: false},
};

module.exports = definition;

```

- save this as file next to `configuration.yaml` as `ext_converter.js`
- add it to `configuration.yaml`:
```yaml
external_converters:
  - ext_converter.js
```
- start z2m

**Reply by bentouspam** [reply]
Great, it works fine !!!

![image](https://user-images.githubusercontent.com/101732505/182457278-b795905a-87c4-4190-9002-3470df1e1bbd.png)

**Reply by piotrbielaska** [reply]
Hi. Is there a chance to add following version as well? _TZ3210_0zabbfax. Zigbee2mqtt says it's not supported.

**Reply by Koenkk** [reply]
@piotrbielaska added!

Changes will be available in the dev branch in a few hours from now. (https://www.zigbee2mqtt.io/advanced/more/switch-to-dev-branch.html)

**Reply by ptv1p3r** [reply]
> Hi. Is there a chance to add following version as well? _TZ3210_0zabbfax. Zigbee2mqtt says it's not supported.

Exactly my problem also with _TZ3210_0zabbfax

**Reply by ptv1p3r** [reply]
I've made the changes and added the files and it works great, one question do i need to delete the file and remove the configuration entry on configuration.yaml when the next release is deployed that allready contains the _TZ3210_0zabbfax ??

**Reply by ks09aao** [reply]
I have tried with and without the external converter above (after adding a row entry to it) and still can't get my _TZ3210_0zabbfax to work. 

@ptv1p3r how did you get yours to work? 

![image](https://user-images.githubusercontent.com/69645797/204492850-c1c6e3a0-7d3a-4f28-92c3-bcc2355cfba4.png)

**Reply by ptv1p3r** [reply]
First i added the device to zigbee2mqtt the normal way, click on allow to join devices and the on device pressed to join, z2mqtt added the device but stated it's not yet recognized, then i added the file external_decoder and added a new entry ( {modelID: 'TS0503B', manufacturerName: '_TZ3210_0zabbfax'}), after i added the external_decoded entry to the configuration.yaml and restarted z2mqqt.

**Reply by ks09aao** [reply]
> First i added the device to zigbee2mqtt the normal way, click on allow to join devices and the on device pressed to join, z2mqtt added the device but stated it's not yet recognized, then i added the file external_decoder and added a new entry ( {modelID: 'TS0503B', manufacturerName: '_TZ3210_0zabbfax'}), after i added the external_decoded entry to the configuration.yaml and restarted z2mqqt.

That's what I tried too - just tried again and still says unsupported. Maybe the issue is with my decoder.js - could I ask you to share yours?

**Reply by ptv1p3r** [reply]
`const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');
const ota = require('zigbee-herdsman-converters/lib/ota');
const tuya = require('zigbee-herdsman-converters/lib/tuya');
const e = exposes.presets;
const ea = exposes.access;

const definition = {
    fingerprint: [{modelID: 'TS0503B', manufacturerName: '_TZ3000_i8l0nqdu'},
        {modelID: 'TS0503B', manufacturerName: '_TZ3210_a5fxguxr'},
        {modelID: 'TS0503B', manufacturerName: '_TZ3210_778drfdt'},
        {modelID: 'TS0503B', manufacturerName: '_TZ3000_g5xawfcq'},
        {modelID: 'TS0503B', manufacturerName: '_TZ3210_trm3l2aw'},
        {modelID: 'TS0503B', manufacturerName: '_TZ3210_95txyzbx'},
        {modelID: 'TS0503B', manufacturerName: '_TZ3210_odlghna1'},
        {modelID: 'TS0503B', manufacturerName: '_TZ3210_0zabbfax'}],
    model: 'TS0503B',
    vendor: 'TuYa',
    description: 'Zigbee RGB light',
    extend: extend.light_onoff_brightness_color(),
    // Requires red fix: https://github.com/Koenkk/zigbee2mqtt/issues/5962#issue-796462106
    meta: {applyRedFix: true, enhancedHue: false},
};

module.exports = definition;`

**Reply by ks09aao** [reply]
working now - i was restarting my home assistant OS, did not realize I should restart MQTT! Thanks for your help

**Reply by ptv1p3r** [reply]
Add this : 
external_converters:
  - ext_converter.js
  
To the configuration.yaml before the devices entry, and make sure that the external_converter file is next to the configuration.yaml file
